OmsiHook/CustomAttributes.cs

```diff
@@ -188,29 +198,39 @@ public OmsiStructArrayPtrAttribute(Type objType, Type internalType = null)
             this.objType = objType;
             this.requiresExtraMarshalling = internalType != null;
             this.internalType = internalType ?? objType;
+            this.raw = raw;
         }
 
         public Type ObjType => objType;
         public Type InternalType => internalType;
         public bool RequiresExtraMarshalling => requiresExtraMarshalling;
+        public bool Raw => raw;
     }
 
     /// <summary>
     /// Marks a field to be converted from an <c>int</c> to an array of strings.<para/>
     /// Used by <seealso cref="Memory.MarshalStruct{OutStruct, InStruct}(InStruct)"/>
     /// </summary>
     [System.AttributeUsage(AttributeTargets.Field, Inherited = false, AllowMultiple = false)]
-    sealed class OmsiStrArrayPtrAttribute : Attribute
+    sealed class OmsiStrArrayPtrAttribute : OmsiMarshallerAttribute
     {
         // See the attribute guidelines at 
         //  http://go.microsoft.com/fwlink/?LinkId=85236
         readonly bool wide;
+        readonly bool raw;
 
-        public OmsiStrArrayPtrAttribute(bool wide = false)
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="wide"></param>
+        /// <param name="raw">If <see langword="true"/>, treat the <c>address</c> as the pointer to the first element 
+        /// of the array instead of as a pointer to the array.</param>
+        public OmsiStrArrayPtrAttribute(bool wide = false, bool raw = false)
         {
             this.wide = wide;
```

Comment:
Assign to raw here

Reply:
Tbh, my mistake for forgetting to do it earlier.

Reply:
And check that it doesn't break anything else when doing so. I think a few places assumed it was being set to true when it wasn't.